service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -1640,4 +1642,23 @@ public final ConnectionLimitManager getConnectionLimitManager() {
     public final void setConnectionLimitManager(final ConnectionLimitManager connectionLimitManager) {
         this.connectionLimitManager = connectionLimitManager;
     }
+
+    /**
+     * Applies the sampling priority configured for the given tenant and device to the given span.
+     * <p>
+     * The sampling priority defines in how far <em>OpenTracing</em> spans created when processing
+     * messages for the tenant and device shall be recorded (sampled) by the tracing system.
+     *
+     * @param tenantObject The tenant configuration to get the sampling priority information from.
+     * @param deviceId The device id (may be null).
+     * @param span The span to apply the sampling priority to.
+     */
+    protected void applyTraceSamplingPriority(final TenantObject tenantObject, final String deviceId, final Span span) {
```

Comment:
is this supposed to be overridden by subclasses?

Reply:
No, I've marked the method as `final`.